packages/shared/src/features/onboarding/shared/DiscountTimer.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import type { ReactElement } from 'react';
+import classNames from 'classnames';
+import createDOMPurify from 'dompurify';
+
+/**
+ * Formats seconds to MM:SS format
+ */
+const formatTime = (seconds: number): string => {
+  if (seconds <= 0) {
+    return '00:00';
+  }
+
+  const minutes = Math.floor(seconds / 60);
+  const remainingSeconds = seconds % 60;
+
+  return `${minutes.toString().padStart(2, '0')}:${remainingSeconds
+    .toString()
+    .padStart(2, '0')}`;
+};
+
+export interface DiscountTimerProps {
+  discountMessage: string;
+  durationInMinutes: number;
+  startDate?: Date;
+  className?: string;
+  onTimerEnd?: () => void;
+}
+
+const calculateTimeLeft = (
+  startDate: Date,
+  durationInMinutes: number,
+): number => {
+  const now = new Date();
+  const endTime = new Date(startDate.getTime() + durationInMinutes * 60 * 1000);
```

Comment:
We could use `addMinutes()` from `date-fns` that we already have in apps 👀

Reply:
Done